Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -116,6 +124,58 @@ private func decodeString(_ s: String) -> String? {
             out.append(n)
             bytes = savedPosition
           }
+        case asciiLowerU, asciiUpperU: // "u"
+          // \u - 4 hex digits, \U 8 hex digits:
+          if let digit1 = bytes.next(),
+              let d1 = uint32FromHexDigit(digit1),
+              let digit2 = bytes.next(),
+              let d2 = uint32FromHexDigit(digit2),
+              let digit3 = bytes.next(),
+              let d3 = uint32FromHexDigit(digit3),
+              let digit4 = bytes.next(),
+              let d4 = uint32FromHexDigit(digit4) {
+            var codePoint = (d1 << 12) + (d2 << 8) + (d3 << 4) + d4
```

Comment:
Personally, I might factor out the process of grabbing the next four bytes by using a local function that accepted the iterator.  But I might not.  😉   Up to you.

Reply:
Done.  Inserted a new commit to do it so the diff is pretty easy to look at.